chain/client/src/sync/state.rs

```diff
@@ -702,6 +702,7 @@ impl StateSync {
         )?;
 
         if all_done {
+            chain.process_snapshot_after_resharding()?;
```

Comment:
Would it make more sense to call this in `build_state_for_split_shards_postprocessing`? Why tag into the whole state sync process, which may run even if there's no resharding happening.

Reply:
yeah I wanted to but postprocessing is per shard

Reply:
Oops, you're right! Apologies, ignore my comment!